libs/protocol-client/src/lib/client/contribution.ts

```diff
@@ -82,6 +85,74 @@ export class Contribution extends BaseClient {
     return contributions.updateContribution;
   }
 
+  public async bulkMint(
+    networkConfig: NetworkConfig,
+    signer: ethers.Signer,
+    address: string,
+    contributions: {
+      id: number;
+      activityTypeId: number;
+      userId: number;
+      args: MintArgs;
+      name: Uint8Array;
+      details: Uint8Array;
+      proof: Uint8Array;
+    }[],
+  ) {
+    const contract = new GovrnContract(networkConfig, signer);
+    const args = {
+      contributions: contributions.map(c => ({
+        contribution: {
+          owner: address,
+          detailsUri: c.args.detailsUri,
+          dateOfSubmission: c.args.dateOfSubmission,
+          dateOfEngagement: c.args.dateOfEngagement,
+        },
+      })),
+    };
+    const transaction = await contract.bulkMint(args);
+    const transactionReceipt = await transaction.wait(REQUIRED_CONFIRMATIONS);
+
+    const onChainIds: { [index: number]: ethers.BigNumber | null } =
+      transactionReceipt.logs
```

Comment:
Why not do this in a single for loop? Doesn't map and reduce loop over all the objects in the array twice?